_posts/2018-02-16-running-foreman-with-postgresql-96-from-software-collections.md

```diff
@@ -0,0 +1,90 @@
+
+Looking at [`postgresql::globals`](https://github.com/puppetlabs/puppetlabs-postgresql/blob/master/manifests/globals.pp) and [`postgresql::params`](https://github.com/puppetlabs/puppetlabs-postgresql/blob/master/manifests/params.pp), we come up with the following settings:
+
+```yaml
+postgresql::params::version: 9.6
+postgresql::globals::client_package_name: rh-postgresql96-postgresql-syspaths
+postgresql::globals::server_package_name: rh-postgresql96-postgresql-server-syspaths
+postgresql::globals::contrib_package_name: rh-postgresql96-postgresql-contrib-syspaths
+postgresql::globals::service_name: rh-postgresql96-postgresql
+postgresql::globals::datadir: /var/opt/rh/rh-postgresql96/lib/pgsql/data
+postgresql::globals::confdir: /var/opt/rh/rh-postgresql96/lib/pgsql/data
+postgresql::globals::bindir: /usr/bin
+```
+
+Just add these lines to `/etc/foreman-installer/custom-hiera.yaml` before running `foreman-installer` and we end up with Foreman running against a PostgreSQL 9.6.
+
+There is one small issue with this setup, though. For some reason `/var/opt/rh/rh-postgresql96/lib/pgsql/data/postgresql.conf` is not getting the correct SELinux context when created by Puppet, and thus PostgreSQL fails to start. Thus you'd need to either run the Installer in permissive mode and relabel the file afterwards (`restorecon -Rv /var/opt/rh/rh-postgresql96/lib/pgsql/data/postgresql.conf`), or expect the failure, relabel and re-run the installier a second time.
+
+Don't worry that we see also PostgreSQL 9.2 packages in `rpm -qa`:
+
+```
+# rpm -qa |grep postgresql
+foreman-postgresql-1.16.0-1.el7.noarch
+rh-postgresql96-postgresql-9.6.5-1.el7.x86_64
+rh-postgresql96-postgresql-libs-9.6.5-1.el7.x86_64
+rh-postgresql96-postgresql-syspaths-9.6.5-1.el7.x86_64
+rh-postgresql96-postgresql-server-syspaths-9.6.5-1.el7.x86_64
+postgresql-libs-9.2.23-3.el7_4.x86_64
+rh-postgresql96-runtime-3.0-10.el7.x86_64
+rh-postgresql96-postgresql-server-9.6.5-1.el7.x86_64
+```
+
+`foreman-postgresql` requires `tfm-rubygem-pg` which is compiled against `postgresql-libs`, and that's totally fine, as this is only the client parts for Ruby that are running with an older `libpq`.
+
+## quick setup with forklift
+
+If you just want to try it out, and love [`forklift`](https://github.com/theforeman/forklift/) as I do, just get a version of `forklift` with my [`custom-hiera.yaml` PR](https://github.com/theforeman/forklift/pull/613) merged and create a box like this:
+
+```
+$ cat boxes.d/99-local.yaml
+---
+centos7-foreman-1.16-psql96:
+  box: centos7
+  ansible:
+    playbook: playbooks/foreman.yml
+    variables:
+      foreman_repositories_version: 1.16
```

Comment:
I had to add: _foreman_installer_options: "--skip-puppet-version-check"_

Reply:
As discussed on IRC, this is most probably an error on your side, as it seems you had the wrong playbooks referenced. Please report back if the vanilla one runs fine.

Reply:
ack, ignore my comment